chained/water_version_control.go

```diff
@@ -0,0 +1,274 @@
+		}
+
+		files[filename] = wasmInfo{
+			version:   splitFilename[0],
+			protocol:  splitFilename[1],
+			builtWith: splitFilename[2],
+			path:      path,
+		}
+		return nil
+	})
+
+	if err != nil {
+		return nil, log.Errorf("failed to walk through the directory %s: %v", dir, err)
+	}
+
+	return files, nil
+}
+
+// GetWASM returns the WASM file for the given transport
+// Remember to Close the io.ReadCloser after using it
+func (vc *versionControl) GetWASM(ctx context.Context, transport string, urls []string) (io.ReadCloser, error) {
+	info, ok := vc.wasmFilesAvailable[transport]
+	if !ok {
+		var err error
+		info, err = vc.downloadWASM(ctx, transport, urls)
+		if err != nil {
+			return nil, log.Errorf("failed to download WASM file: %w", err)
+		}
+
+		vc.wasmFilesAvailable[transport] = info
+	}
+
+	f, err := os.Open(info.path)
+	if err != nil {
+		return nil, log.Errorf("failed to open file %s: %w", info.path, err)
+	}
+
+	return f, nil
+}
+
+// Commit will update the history of the last time the WASM file was loaded
+// and delete the outdated WASM files
+func (vc *versionControl) Commit(transport string) error {
+	vc.historyMutex.Lock()
+	defer vc.historyMutex.Unlock()
+	if err := vc.loadHistory(); err != nil {
+		return log.Errorf("failed to load history: %w", err)
+	}
+
+	if err := vc.updateLastTimeLoaded(transport); err != nil {
+		return log.Errorf("failed to update last time loaded: %w", err)
+	}
+
+	if err := vc.deleteOutdatedWASMFiles(); err != nil {
+		return log.Errorf("failed to delete outdated WASM files: %w", err)
+	}
+	return nil
+}
+
+func (vc *versionControl) updateLastTimeLoaded(transport string) error {
+	for i, h := range vc.history {
+		if h.Transport == transport {
+			vc.history[i].LastTimeLoaded = time.Now()
+			if err := vc.storeHistory(); err != nil {
+				return log.Errorf("failed to store history: %w", err)
+			}
+			return nil
+		}
+	}
+
+	vc.history = append(vc.history, history{
+		Transport:      transport,
+		LastTimeLoaded: time.Now(),
+	})
+	if err := vc.storeHistory(); err != nil {
+		return log.Errorf("failed to store history: %w", err)
+	}
+	return nil
+}
+
+func (vc *versionControl) loadHistory() error {
+	historyCSV := filepath.Join(vc.dir, "history.csv")
+
+	// if there's no history available, just initialize history
+	if _, err := os.Stat(historyCSV); os.IsNotExist(err) {
+		vc.history = []history{}
+		return nil
+	}
+
+	f, err := os.Open(historyCSV)
+	if err != nil {
+		return log.Errorf("failed to open file %s: %w", historyCSV, err)
+	}
+	defer f.Close()
+
+	history := []history{}
+	if err := gocsv.UnmarshalFile(f, &history); err != nil {
+		return log.Errorf("failed to unmarshal file %s: %w", historyCSV, err)
+	}
+
+	vc.history = history
+	return nil
+}
+
+func (vc *versionControl) deleteOutdatedWASMFiles() error {
+	affectedTransports := make([]string, 0)
+	for _, h := range vc.history {
+		// check if the file is outdated
+		sevenDaysAgo := time.Now().AddDate(0, 0, -7)
```

Comment:
Could you move this constant to the top of the file and make it clear what it's used for?  Something like:
```
const unusedWASMsDeletedAfter = 7 * 24 * time.Hour
```